Xdl.Internship.Core/Xdl.Internship.Core.DataAccess.MongoDB/Repositories/MongoRepository.cs

```diff
@@ -0,0 +1,103 @@
+        {
+            return _collectionAccessor.Value.FindSync(filterExpression, null, cancellationToken).ToListAsync(cancellationToken);
+        }
+
+        public virtual Task<TDocument> TryFindOneAsync(Expression<Func<TDocument, bool>> filterExpression, CancellationToken cancellationToken = default)
+        {
+            return _collectionAccessor.Value.Find(filterExpression).FirstOrDefaultAsync(cancellationToken);
+        }
+
+        public virtual Task<TDocument> FindByIdAsync(ObjectId id, CancellationToken cancellationToken = default)
+        {
+            var filter = _fdb.Eq(doc => doc.Id, id);
+            return _collectionAccessor.Value.Find(filter).SingleOrDefaultAsync(cancellationToken);
+        }
+
+        // Create
+        public virtual Task InsertOneAsync(TDocument document, CancellationToken cancellationToken = default)
+        {
+            return _collectionAccessor.Value.InsertOneAsync(document, null, cancellationToken);
+        }
+
+        public virtual async Task InsertManyAsync(ICollection<TDocument> documents, CancellationToken cancellationToken = default)
+        {
+            await _collectionAccessor.Value.InsertManyAsync(documents, null, cancellationToken);
+        }
+
+        // Update (put)
+        public virtual async Task ReplaceOneAsync(TDocument document, CancellationToken cancellationToken = default)
+        {
+            var filter = _fdb.Eq(doc => doc.Id, document.Id);
+            await _collectionAccessor.Value.FindOneAndReplaceAsync(filter, document, cancellationToken: cancellationToken);
+        }
+
+        // Delete
+        public virtual Task DeleteOneAsync(Expression<Func<TDocument, bool>> filterExpression, CancellationToken cancellationToken = default)
+        {
+            return _collectionAccessor.Value.FindOneAndDeleteAsync(filterExpression, cancellationToken: cancellationToken);
+        }
+
+        public virtual Task DeleteByIdAsync(ObjectId id, CancellationToken cancellationToken)
+        {
+            var filter = _fdb.Eq(doc => doc.Id, id);
+            return _collectionAccessor.Value.FindOneAndDeleteAsync(filter, cancellationToken: cancellationToken);
+        }
+
+        public virtual Task DeleteManyAsync(Expression<Func<TDocument, bool>> filterExpression, CancellationToken cancellationToken = default)
+        {
+            return _collectionAccessor.Value.DeleteManyAsync(filterExpression, cancellationToken);
+        }
+
+        // Read
+        public virtual async Task<ICollection<TDocument>> GetAsync(ObjectId id, CancellationToken cancellationToken = default)
+        {
+            var filter = _fdb.Eq(doc => doc.Id, id);
+            return await FindAsync(filter, null, cancellationToken);
+        }
+
+        public virtual async Task<ICollection<TDocument>> GetAllAsync(CancellationToken cancellationToken = default)
+        {
+            return await FindAsync(_fdb.Empty, null, cancellationToken);
```

Comment:
remove async and await

Reply:
what a difference between this method and FindByIdAsync?

Reply:
remove duplicate

Reply:
resolved(removed)